rust-runtime/smithy-eventstream/src/frame.rs

```diff
@@ -94,8 +97,12 @@ mod value {
                 }
                 TYPE_TIMESTAMP => {
                     if buffer.remaining() >= size_of::<i64>() {
-                        Ok(HeaderValue::Timestamp(Instant::from_epoch_seconds(
-                            buffer.get_i64(),
+                        let time = buffer.get_i64();
+                        let seconds = time / 1000;
+                        let millis = time.rem_euclid(1000) as u32;
+                        Ok(HeaderValue::Timestamp(Instant::from_secs_and_nanos(
+                            seconds,
+                            millis * NANOS_PER_MILLI_U32,
```

Comment:
I'd go through the stdlib here:
```rust
let time = buffer.get_i64();
let dur = Duration::from_millis(time.abs() as u64);
let instant = Instant::from_secs_and_nanos(dur.as_secs() as i64 * time.signum(), dur.subsec_nanos());
```

Adding a `from_millis(i64)` to instant would also be totally reasonable